packages/idyll-document/src/index.js

```diff
@@ -30,48 +31,14 @@ const getRefs = () => {
   if (!scrollContainer) {
     return;
   }
-  const containerNode = scrollContainer.item;
-  const containerRect = containerNode.getBoundingClientRect();
 
   scrollWatchers.forEach(watcher => {
-    // get boolean props
-    let bools = {};
-    Object.keys(watcher).forEach(key => {
-      if (!key.startsWith('is')) return;
-      bools[key] = watcher[key];
-    });
-
-    const domNode = watcher.watchItem;
-    const rect = domNode.getBoundingClientRect();
-    const portionInView = Math.max(
-      0,
-      Math.min(
-        1,
-        (watcher.bottom - scrollContainer.viewportTop) / rect.height
-      )
-    )
-
     // left and right props assume no horizontal scrolling
+    const { watchItem, callbacks, container, recalculateLocation, offsets, ...watcherProps} = watcher;
+    const domNode = watchItem;
     refs[domNode.dataset.ref] = {
-      ...bools,
-      domNode,
-      portionInView,
-      size: {
-        width: rect.width,
-        height: rect.height,
-      },
-      position: {
-        top: watcher.top - scrollContainer.viewportTop,
-        right: rect.right - containerRect.left,
-        bottom: watcher.bottom - scrollContainer.viewportTop,
-        left: rect.left - containerRect.left,
-      },
-      absolutePosition: {
-        top: watcher.top,
-        right: rect.right,
-        bottom: watcher.bottom,
-        left: rect.left,
-      },
+      ...watcherProps,
+      domNode
```

Comment:
I'd probably just do `domNode: watchItem` here instead of creating the extra `const`.